tests/e2e/cmp_test.go

```diff
@@ -55,7 +55,8 @@ var _ = Describe("odoCmpE2e", func() {
 		})
 
 		It("should show an error when ref flag is provided with sources except git", func() {
-			output := runFailCmd(fmt.Sprintf("odo create nodejs cmp-git-%s --local test --ref test", t))
+			output, _ := runFailCmd(fmt.Sprintf("odo create nodejs cmp-git-%s --local test --ref test", t))
+
 			Expect(output).To(ContainSubstring("The --ref flag is only valid for --git flag"))
```

Comment:
Here return of the command is captured in stderr not in output causing CI failure.  

Try something like 
_, stderr := runFailCmd(fmt.Sprintf("odo create nodejs cmp-git-%s --local test --ref test", t))
Expect(fmt.Fprintf(stderr)).To(ContainSubstring("The --ref flag is only valid for --git flag"))

Reply:
As pr @girishramnani comments https://github.com/redhat-developer/odo/pull/1182/files#r248625658 and https://github.com/redhat-developer/odo/pull/1182/files#r248626040 the above comment will not resolve the failure i guess